test-framework/junit5-internal/src/main/java/io/quarkus/test/QuarkusProdModeTest.java

```diff
@@ -138,16 +137,10 @@ public class QuarkusProdModeTest
     private boolean clearRestAssuredURL;
 
     public QuarkusProdModeTest() {
-        InputStream appPropsIs = Thread.currentThread().getContextClassLoader().getResourceAsStream("application.properties");
-        if (appPropsIs != null) {
-            customApplicationProperties = new Properties();
-            try (InputStream is = appPropsIs) {
-                customApplicationProperties.load(is);
-            } catch (IOException e) {
-                throw new UncheckedIOException("Failed to load application configuration from "
-                        + Thread.currentThread().getContextClassLoader().getResource("application.properties"), e);
-            }
-        }
+        // If there is an application.properties resource then load the properties unless a custom config resource name is used
+        this.configResourceName = Thread.currentThread().getContextClassLoader().getResource(APPLICATION_PROPERTIES) != null
+                ? APPLICATION_PROPERTIES
+                : null;
```

Comment:
Since we pick `application.properties` from the classpath by default, and you fixed a problem that prevented it from overriding resources that already exist in the archive... Aren't we introducing a bug here, with this "default override"?

In particular, when I do this:

```java
    @RegisterExtension
    static final QuarkusProdModeTest TEST = new QuarkusProdModeTest()
            .withApplicationRoot((jar) -> jar
                    .addClasses(MyEntity.class)
                    .addAsResource("application-custom.properties", "application.properties")
                    .addAsResource("import.sql"));
```

... then if there is an `application.properties` in the classpath, it will erase the one I just added to my archive, will it not? :thinking:

Reply:
> ... then if there is an application.properties in the classpath, it will erase the one I just added to my archive, will it not? 🤔

Uf, you're right. But I wanted to keep this behavior so that we don't break the existing tests. What we could do is to only load the properties if `withConfigurationResource()` is not used && no `application.properties` is present in the test archive.

Reply:
> Uf, you're right. But I wanted to keep this behavior so that we don't break the existing tests

Right, my point was that, combined with you fixing the "resource precedence", this will just break existing tests in a different way :)

> What we could do is to only load the properties if `withConfigurationResource()` is not used && no `application.properties` is present in the test archive.

Looks like a plan :+1: